cpp/src/utilities/cuda_utils.hpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2019 BlazingDB, Inc.
+ *     Copyright 2019 Eyal Rozenberg <eyalroz@blazingdb.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CUDF_UTILITIES_CUDA_UTIL_CUH_
+#define CUDF_UTILITIES_CUDA_UTIL_CUH_
+
+/**
+ * @file Utility code for host-side interaction with CUDA by cuDF API
+ * function implementations
+ */
+
+#include <cuda_runtime_api.h>
+
+namespace cudf {
+
+namespace util {
+
+namespace cuda {
+
+/**
+ * A RAII gadget for top-level CUDF API function implementations, which may
+ * create their own streams. This will ensure the stream is destroyed when the API function returns,
+ * without you having to explicitly destroy it - and also when you exit due to an error.
+ */
+struct scoped_stream {
+    cudaStream_t stream_ { nullptr };
+
+    scoped_stream() {
+        cudaStreamCreate(&stream_);
```

Comment:
Wrap with `CUDA_TRY` and the below `assert` can be removed.

Reply:
Done.